cryptoutils/crypto_test.go

```diff
@@ -0,0 +1,712 @@
+package cryptoutils
```

Comment:
Amazing work on test cases 🚀 
I'll leave the main review to @shahzadhaider1 - If he approves I'll merge.

Reply:
LGTM @kashifkhan0771